cloud/docker/docker.py

```diff
@@ -796,13 +804,22 @@ def get_differing_containers(self):
                     continue
 
             # EXPOSED PORTS
-            # Note that ports that are bound at container run are also exposed
-            # implicitly.
-            expected_exposed_ports = set()
+
+            image_exposed_ports = set((image["ContainerConfig"]["ExposedPorts"] or {}).keys())
+
+            expected_exposed_ports = set(image_exposed_ports)
             for p in (self.exposed_ports or []):
                 expected_exposed_ports.add("/".join(p))
 
-            actually_exposed_ports = set((container["Config"]["ExposedPorts"] or {}).keys())
+            if self.port_bindings:
+                for container_port in self.port_bindings.keys():
+                    if isinstance(container_port, int):
+                        container_port = "{}/tcp".format(container_port)
+
+                    expected_exposed_ports.add(container_port)
```

Comment:
Uh, kind of? It seems a lot easier to me to only check explicitly exposed ports in the exposed ports section. What does adding `self.port_bindings` actually gain?

I'm playing with this myself right now in my test playbook, let's see if I can put something together.